temporalio/worker/_workflow_instance.py

```diff
@@ -158,6 +159,16 @@ def activate(
         """
         raise NotImplementedError
 
+    def get_thread_id(self) -> Optional[int]:
+        """Return the thread identifier that this workflow is running on.
+
+        Not an abstractmethod because it is not mandatory to implement. Used primarily for getting the frames of a deadlocked thread.
+
+        Returns:
+            Thread ID if the workflow is running, None if not.
+        """
+        return None
```

Comment:
Can I just check -- is there a reason for this getter? Do we want people to have the ability to create `Workflow` classes that refuse to honor `self._current_thread_id`? Or would it just work to use `self._current_thread_id` directly everywhere? (Python doesn't use getters unless necessary).